compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/MainMethodGen.java

```diff
@@ -350,7 +349,6 @@ private void createOption(MethodVisitor mv, BIRNode.BIRFunctionParameter param,
         mv.visitTypeInsn(NEW , OPTION);
         mv.visitInsn(DUP);
         jvmTypeGen.loadType(mv, param.type);
-        mv.visitTypeInsn(CHECKCAST , RECORD_TYPE);
```

Comment:
Don't we need to have the `checkcast` instruction here since the `param.type` is an instance of RecordType?

Reply:
We should not check the cast here, because the function parameter type can be a `TypeReferenceType` now. 
```ballerina
public type Student record {|
    string name;
|};

public type S Student;

public function main(*S student) {
// 
}
```